plugins/woocommerce/src/Internal/Admin/Logging/FileV2/File.php

```diff
@@ -0,0 +1,173 @@
+<?php
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Internal\Admin\Logging\FileV2;
+
+/**
+ * File class.
+ *
+ * An object representation of a single log file.
+ */
+class File {
+	/**
+	 * The absolute path of the file.
+	 *
+	 * @var string
+	 */
+	protected $path;
+
+	/**
+	 * The source property of the file, derived from the filename.
+	 *
+	 * @var string
+	 */
+	protected $source = '';
+
+	/**
+	 * The 0-based increment of the file, if it has been rotated. Derived from the filename. Can only be 0-9.
+	 *
+	 * @var int|null
+	 */
+	protected $rotation;
+
+	/**
+	 * The date the file was created, as a Unix timestamp, derived from the filename.
+	 *
+	 * @var int|false
+	 */
+	protected $created = false;
+
+	/**
+	 * The hash property of the file, derived from the filename.
+	 *
+	 * @var string
+	 */
+	protected $hash = '';
+
+	/**
+	 * Class File
+	 *
+	 * @param string $path The absolute path of the file.
+	 */
+	public function __construct( $path ) {
+		$this->path = $path;
+		$this->parse_filename();
+	}
+
+	/**
+	 * Parse the log filename to derive certain properties of the file.
+	 *
+	 * This makes assumptions about the structure of the log file's name. Using `-` to separate the name into segments,
+	 * if there are at least 5 segments, it assumes that the last segment is the "key" (i.e. hash), and the three
+	 * segments before that make up the date when the file was created in YYYY-MM-DD format. Any segments left after
+	 * that are the "source" that generated the log entries. If the filename doesn't have enough segments, it falls
+	 * back to the source and the key both being the entire filename, and using the last modified time as the creation
+	 * date.
+	 *
+	 * @return void
+	 */
+	protected function parse_filename(): void {
+		$info     = pathinfo( $this->path );
+		$filename = $info['filename'];
+		$segments = explode( '-', $filename );
+
+		if ( count( $segments ) >= 5 ) {
+			$this->source  = implode( '-', array_slice( $segments, 0, -4 ) );
+			$this->created = strtotime( implode( '-', array_slice( $segments, -4, 3 ) ) );
+			$this->hash    = array_slice( $segments, -1 )[0];
+		} else {
+			$this->source  = implode( '-', $segments );
+			$this->created = filemtime( $this->path );
```

Comment:
```suggestion
			$this->created = filectime( $this->path );
```

Probably not essential since this is a fallback for irregularly named files (and on some systems it doesn't seem to yield much of a difference in any case) but I wonder if this would be closer to the likely creation time (it would definitely reflect the creation time on Windows, and may be closer in practice on some other systems).